src/groth16/prover.rs

```diff
@@ -278,33 +309,110 @@ where
             .collect::<Vec<_>>(),
     );
 
-    let l = multiexp(
-        &worker,
-        params.get_l(aux_assignment.len())?,
-        FullDensity,
-        aux_assignment.clone(),
-        &mut multiexp_kern,
-    );
+    // Keep checking between multiexp
+    #[cfg(feature = "gpu")]
```

Comment:
There is a lot of repetition here, can extract into either a macro or a function please